Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/WatchManager.java

```diff
@@ -474,6 +650,84 @@ private void updateState(boolean watchEnabled, boolean backgroundEnabled) {
         }
     }
 
+    private void updateDeletedOrArchivedPins() {
+        for (Pin pin : pins) {
+            if (pin.isError || pin.archived || !pin.watching) {
+                SavedThread savedThread = findSavedThreadByLoadableId(pin.loadable.id);
+                if (savedThread == null || savedThread.isStopped || savedThread.isFullyDownloaded) {
+                    continue;
+                }
+
+                if (pin.isError || pin.archived) {
+                    savedThread.isFullyDownloaded = true;
+                }
+
+                if (!savedThread.isFullyDownloaded) {
+                    continue;
+                }
+
+                // TODO: should the pinType be updated here?
```

Comment:
Meaning the pinType of the cached pin (the pins list) not the one in the DB (it gets updated below).

Reply:
Still have no idea whether should it be updated or not.